Applications/SlicerApp/Testing/Python/JRC2013Vis.py

```diff
@@ -319,19 +304,15 @@ def test_Part2Head(self):
     #
     # first, get some data
     #
-    import urllib
-    downloads = (
-        ('http://slicer.kitware.com/midas3/download?items=8609', '3DHeadData.mrb', slicer.util.loadScene),
+    from SampleData import SampleDataSource, SampleDataLogic
+    dataSources = [
```

Comment:
Most of the time we download a single data source, so creating a list of sources is unnecessary complication. Could we change the code to this simpler pattern?

```
from SampleData import SampleDataSource, SampleDataLogic
SampleDataLogic().downloadFromSource(SampleDataSource(
  fileNames='3DHeadData.mrb', 
  uris='http://slicer.kitware.com/midas3/download?items=8609', 
  loadFileType='SceneFile'))
```

Reply:
Agreed. Just wanted to get some feedback ... before simplifying.

What about that then:

```
SampleDataLogic().download(
  fileNames='3DHeadData.mrb', 
  uris='http://slicer.kitware.com/midas3/download?items=8609', 
  loadFileType='SceneFile')
```

Reply:
Also we call the function `downloadFromSource` (or `download`) but most of the time it is loaded in Slicer.

Should instead of the method be named
* `SampleData.loadFromSource` or `SampleData.downloadAndLoadFromSource`
* `SampleData.load` or `SampleData.downloadAndLoad`

Reply:
If the input of the method is not a `DataSource` object then we don't need to include that in the method name. So, `SampleData.load` would work well.

Probably `SampleData.loadData` or ``SampleData.loadFromURL` would be even better because it would not suggest that the input is some sample data.

`SampleData.load` could be used for loading sample data, such as `SampleData.load('MRHead')`.

Reply:
What do you think about specifying a helper function at the module level to further simplify syntax?

```
import SampleData
SampleData.loadFromURL(…)
```

Reply:
Or we could have methods like this directly in SampleData:

```
def loadFromURL(...):
  return SampleDataLogic().loadFromURL()

def downloadMRHead():
  return SampleDataLogic().downloadMRHead()

[...]
```

Reply:
That said, I just realized that `downloadMRHead` and a like are deprecated. And that the new modern approach is to use `downloadSample("MRHead")`, it means your suggestion would work very well